proto/substrait/algebra.proto

```diff
@@ -267,11 +267,25 @@ message CrossRel {
 message FetchRel {
   RelCommon common = 1;
   Rel input = 2;
-  // the offset expressed in number of records
-  int64 offset = 3;
-  // the amount of records to return
-  // use -1 to signal that ALL records should be returned
-  int64 count = 4;
+  oneof offset_mode {
+    // the offset expressed in number of records
+    // Deprecated: use `offset_expr` instead
+    int64 offset = 3 [deprecated = true];
+    // Expression evaluated into an integer specifying the number of records to
+    // skip.
+    // Recommended type for offset is int64.
+    Expression offset_expr = 5;
+  }
+  oneof count_mode {
+    // the amount of records to return
+    // use -1 to signal that ALL records should be returned
+    // Deprecated: use `count_expr` instead
+    int64 count = 4 [deprecated = true];
```

Comment:
As I've thought about this more, I don't like that the "not set" state is now inside an expression (e.g. -1 and 0).

I suggest we add another branch to each of these that is 

`google.protobuf.Empty no_count = 7`

and

`google.protobuf.Empty no_offset = 6`

or similar. My other thought is we could just keep the existing fields as not-deprecated (oneof is sufficient) and clearly state that the -1 behavior isn't available to count_expr.

Reply:
How about the expression returns a nullable int64?  Presumably the expression needs to have the same expressibility as the non-expression forms.

That wouldn't change the not set behavior for count though.

Reply:
@jacques-n , I was looking into how `oneof` works in protobuf and I think your concern might already be addressed in the current PR. The unset state is handled at the `oneof` level, i.e., a `oneof` field is always optional so you can just check at the top level whether it's set. So based on that, it seems that `no_count`/`no_offset` would be redundant, since when oneof is unset, it already kind of works like `no_count`/`no_offset`. Regarding the other question, a field within `oneof` cannot be optional (I assume this is what you meant by nullable @EpsilonPrime?, protobuf doesn't allow an optional modifier for `oneof` fields.

Reply:
> would be redundant

Good point. Maybe add a comment in docs clarifying that this is the expected pattern. It's a bit non-obvious (especially given how non-oneof int fields work)

Reply:
Currently the guidance for what type count_expr returns is int64.  Also -1 means returns all records.  If we changed the guidance for the count_expr rexpression to return a nullable int64 we could have null mean no filter/count limit.

Reply:
@EpsilonPrime , I have updated the documentation. Please let me know if this is what you had in mind.